tests/testthat/test-exercise.R

```diff
@@ -557,7 +557,81 @@ test_that("error if env var directory does not exist", {
     ex <- mock_exercise(user_code = 'readLines("data/test.txt")')
     expect_error(
       evaluate_exercise(ex, new.env(), evaluate_global_setup = TRUE),
-      class = "learnr.missing_source_dir"
+      class = "learnr_missing_source_data_dir"
     )
   })
 })
+
+test_that("alternate data directory specified with `options()`", {
+  withr::with_tempdir({
+    dir.create("data")
+    writeLines("DEFAULT", "data/test.txt")
+    dir.create("nested/structure/data", recursive = TRUE)
+    writeLines("NESTED", "nested/structure/test.txt")
+
+    ex <- mock_exercise(
+      user_code = 'readLines("data/test.txt")'
+    )
+    output <- evaluate_exercise(ex, envir = new.env())
+    expect_match(output$html_output, "DEFAULT", fixed = TRUE)
```

Comment:
A little trick that we didn't talk about (because I forgot about it until I read through a few tests) is that you can set `check = TRUE` in `mock_exercise()` this uses a debugging exercise checker that returns a value in the `$feedback` item of the exercise result. 

In that flow, you can access any of the objects that were made available to the exercise checker function via `$feedback$checker_args`. So, for example, you can directly access the result of running the user code like this:

```suggestion
    ex <- mock_exercise(
      user_code = 'readLines("data/test.txt")',
      check = TRUE
    )
    res <- evaluate_exercise(ex, envir = new.env())
    expect_equal(res$feedback$checker_args$last_value, "DEFAULT")
```

You can use this trick to run code at various stages of the exercise evaluation that you can then access via the `envir_prep` or `envir_result` environments that are passed to the checker function.